config/ompi_fortran_check_use_only.m4

```diff
@@ -35,24 +37,31 @@ AC_DEFUN([OMPI_FORTRAN_CHECK_USE_ONLY],[
 
     AC_CACHE_CHECK([if Fortran compiler supports USE...ONLY], use_only_var,
        [AC_LANG_PUSH([Fortran])
-        AC_COMPILE_IFELSE([AC_LANG_SOURCE([[MODULE aaa
-INTEGER :: aaa_unique
-COMMON/common_to_both/COMMON_NAME_TO_BOTH
+        cat > aaa.f90 << EOF
+MODULE aaa
+INTEGER :: CMON(1)
+COMMON/CMMON/CMON
+INTEGER :: global_aaa
 END MODULE aaa
-
+EOF
+        cat > bbb.f90 << EOF
 MODULE bbb
-INTEGER :: bbb_unique
-INTEGER, BIND(C, name="common_to_both_") :: COMMON_NAME_TO_BOTH
+integer, bind(C, name="cmmon_") :: CMON
+INTEGER :: global_bbb
 END MODULE bbb
-
-PROGRAM test_proc
-  USE :: aaa, ONLY: aaa_unique
-  USE :: bbb, ONLY: bbb_unique
+EOF
+        OPAL_LOG_COMMAND([$FC $FCFLAGS -c aaa.f90],
+                         [OPAL_LOG_COMMAND([$FC $FCFLAGS -c bbb.f90],
+                                           [AC_COMPILE_IFELSE([AC_LANG_SOURCE([[PROGRAM test
+USE aaa, ONLY : global_aaa
+USE bbb, ONLY : global_bbb
```

Comment:
Is `$FCFLAGS` set to something here that will guarantee to pick up modules in the current directory?